src/components/PopupWithForm.js

```diff
@@ -25,14 +25,16 @@ export default class PopupWithForm extends Popup {
 
     this._popupForm.addEventListener("submit", (event) => {
       event.preventDefault();
-      this._handleFormSubmit(this._getInputValues());
+      const inputValues = this._getInputValues();
+      console.log("Input Values:", inputValues);
+      this._handleFormSubmit(inputValues);
 
-      this.close();
+      // this.close();
     });
   }
 
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`